File: Python/Functional-2/no_yy.py

# given a list of strings, return a list where each string has "y" added at its end, omitting any resulting strings that contain "yy" as a substring anywhere
def no_yy(strings):
    new_list = []

    for x in strings:
        if x[len(x)-1] != 'y':
            new_list.append(x + 'y')

    return new_list

print(no_yy(['a', 'b', 'c']))
print(no_yy(['a', 'b', 'cy']))
print(no_yy(['xx', 'ya', 'zz']))

File: Java/Functional-2/noYY.java
import java.util.Arrays;
import java.util.List;

public class functional2 {
	
	// given a list of strings, return a list where each string has "y" added at its end, omitting any resulting strings that contain "yy" as a substring anywhere
	public List<String> noYY(List<String> strings) {
		
		strings.replaceAll(n -> n + "y");
		strings.removeIf(n -> n.contains("yy"));
		
		return strings;
		
	}

	public static void main(String[] args) {
		
		functional2 o = new functional2();
		
		List<String> l1 = Arrays.asList("a", "b", "c");
		List<String> l2 = Arrays.asList("a", "b", "cy");
		List<String> l3 = Arrays.asList("xx", "ya", "zz");
		
		System.out.println(o.noYY(l1));
		System.out.println(o.noYY(l2));
		System.out.println(o.noYY(l3));

	}

}
